Application/Program.cs

```diff
@@ -0,0 +1,175 @@
+
+    }
+    static void DeleteProduct(InventoryService inventory)
+    {
+        Console.Write("Enter product name: ");
+        string name = Console.ReadLine();
+
+        if (inventory.DeleteProduct(name) == false)
+            Console.WriteLine("Product not found");
+        else
+            Console.WriteLine("Product deleted successfully.");
+    }
+    static void SearchProduct(InventoryService inventory)
+    {
+        Console.Write("Enter product name: ");
+        string name = Console.ReadLine();
+
+        if (inventory.Search(name) == null)
+        {
+            Console.WriteLine("Product not found");
+        }
+        else
+        {
+            Console.WriteLine("Product is found.\nName\tPrice\tQuantity");
+            inventory.SearchProduct(name);
+        }
+            
+    }
+
+
+    static void EditProduct(InventoryService inventory)
+    {
+        Console.Write("Enter product name you want to update: ");
+        string name = Console.ReadLine();       
+
+        if (inventory.Search(name)==null)
+        {
+            Console.WriteLine("Product not found.");
+            return;
+        }
+        Console.WriteLine("Enter new name, price, or quantity: ");
+        string newName = String.Empty;
+
+        while (true)
+        {
+            Console.Write("Enter product name: ");
+            newName = Console.ReadLine();
+            if (newName != string.Empty)
+                break;
+            Console.WriteLine("Please enter the name of product.");
+        }
+
+        double newPrice;
+        while (true)
+        {
+            Console.Write("Enter product price: ");
+            if (double.TryParse(Console.ReadLine(), out newPrice) && newPrice >= 0)
+                break;
+            Console.WriteLine("Invalid input! Please enter a number.");
+        }
+
+        int newQuantity;
+        while (true)
+        {
+            Console.Write("Enter product quantity: ");
+            if (int.TryParse(Console.ReadLine(), out newQuantity) && newQuantity >= 0)
+                break;
+            Console.WriteLine("Invalid input! Please enter a number.");
+        }
+
+        
+        inventory.EditProduct(name, newPrice, newQuantity, newName);
+        Console.WriteLine("Product updated successfully.");
+
+    }
+
+    static void Menu()
+    {
+        Console.WriteLine("\n1. Add Product");
+        Console.WriteLine("2. View Products");
+        Console.WriteLine("3. Edit Product");
+        Console.WriteLine("4. Delete Product");
+        Console.WriteLine("5. Search Product");
+        Console.WriteLine("6. Exit");
+        Console.Write("\nChoose an option: ");
+    }
+    static void Main(string[] args)
+    {
+        InventoryService _inventoryService = new InventoryService();
+
+        Console.WriteLine("Welcome to our Inventory Management System!");
+
+        while (true)
+        {
+            Menu();
+
+            string choice = Console.ReadLine();
+            switch (choice)
```

Comment:
NIT: to reduce un-needed code. This approach is considered more cleaner. 
```suggestion
            switch (Console.ReadLine())
```